examples/step-42/step-42.cc

```diff
@@ -938,26 +944,24 @@ namespace Step42
 
         GridGenerator::hyper_rectangle(triangulation, p1, p2);
 
-        /* boundary_ids:
-
-         */
         Triangulation<3>::active_cell_iterator
         cell = triangulation.begin_active(),
         endc = triangulation.end();
         for (; cell != endc; ++cell)
-          for (unsigned int face = 0; face < GeometryInfo<dim>::faces_per_cell;
-               ++face)
-            {
-              if (cell->face(face)->center()[2] == p2(2))
-                cell->face(face)->set_boundary_id(1);
-              if (cell->face(face)->center()[0] == p1(0)
-                  || cell->face(face)->center()[0] == p2(0)
-                  || cell->face(face)->center()[1] == p1(1)
-                  || cell->face(face)->center()[1] == p2(1))
-                cell->face(face)->set_boundary_id(8);
-              if (cell->face(face)->center()[2] == p1(2))
-                cell->face(face)->set_boundary_id(6);
-            }
+          for (unsigned int face_no = 0; face_no < GeometryInfo<dim>::faces_per_cell;
+               ++face_no)
+            if (cell->face(face_no)->at_boundary())
+              {
+                if (cell->face(face_no)->center()[2] == p2(2))
+                  cell->face(face_no)->set_boundary_id(1);
+                if (cell->face(face_no)->center()[0] == p1(0)
+                    || cell->face(face_no)->center()[0] == p2(0)
+                    || cell->face(face_no)->center()[1] == p1(1)
+                    || cell->face(face_no)->center()[1] == p2(1))
```

Comment:
Comparing with `p[i]` has the advantage that it will continue to work if you change the point coordinates above. So I'm in favor of that.

If you'd prefer using `std::fabs`, then I can do that.